Clarify docs referenced PullbackMemory as a Fixed Thing (#469)
JuliaDiff/ChainRulesCore.jl: docs/src/design/changing_the_primal.md

That now looks much simpler; `pullback_at` only ever has 2 arguments.

One way we could make it nicer to use is by making `PullbackMemory` a callable object.

> Conceptually, for a particular instance of an operation the `PullbackMemory` is a fixed thing: it could be recorded as a hardcoded literal on the the tape.

Reviewer: I'm not sure what's meant by `hardcoded literal on the tape`. Could you elaborate a bit so that I can figure out what's intended?

Author: The idea is that this fixed, during the augmented primal pass.
It is not something that is computed in response to the derivative being provided.
Or updated at any point

Reviewer: Oh I see. @st-- does this explanation clarify things for you? If so, we can figure out how to write this succinctly.

Reviewer 2: i.e., the (augmented) primal pass is the process in which the PullbackMemory becomes fixed, yes? yeah that makes sense

Author: Correct.

Reviewer: Perhaps simply
```suggestion
Conceptually, for a particular evaluation of an operation, the `PullbackMemory` is a fixed linear function.
```
?

Author: the memory isn't a `function` in this part of the example.
The memory is I guess you could say the coefficients of a linear function

Reviewer: Oh okay, fair enough. Well maybe just
```suggestion
Conceptually, for a particular evaluation of an operation, the `PullbackMemory` is fixed.
```
then?
